src/mailing/sender.js

```diff
@@ -9,26 +11,28 @@ const transport = nodemailer.createTransport({
   },
 });
 
-// Function to send a test email (promisified)
-async function sendTestEmail(recipient) {
-  return new Promise((resolve, reject) => {
-    transport.sendMail(
-      {
-        from: process.env.EMAIL_USERNAME,
-        to: recipient,
-        subject: 'Test Email',
-        text: 'Hello, this is a test email!',
-        html: '<p>Hello, this is a test email!</p>',
-      },
-      (err, res) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(res);
-        }
-      }
-    );
+const baseEmail = new Email({
+  message: {
+    from: 'codethedream.practicum.team4@outlook.com',
+  },
+  transport: transport, // Use the transport you created
+  subjectPrefix: process.env.NODE_ENV === 'production' ? '' : '(TEST) ',
+  views: {
+    root: path.resolve('../templates/test'), // this tells email-templates to look for templates starting from this path
+    options: {
+      extension: 'ejs', // this tells email-templates that the file will end tith *.ejs
+    },
+  },
+});
+
+// Function to send a test email
+async function sendTestEmail(to, username) {
+  return baseEmail.send({
+    template: 'test',
+    message: { to },
+    locals: { name: username },
```

Comment:
Your email uses the variable `username`

```
 <p>Hello, <%= username %>, this is a test email.</p>
```

but here in locals we're passing in an object that has a field `name` that has a value of whatever `username` evaluates to. So this will cause an error. We should either change the locals to `{ username: username }` or the EJS template to `<p> Hello, <%= name %>, etc`